src/compiler/commandLineParser.ts

```diff
@@ -405,13 +405,107 @@ namespace ts {
       */
     export function parseConfigFileTextToJson(fileName: string, jsonText: string): { config?: any; error?: Diagnostic } {
         try {
-            return { config: /\S/.test(jsonText) ? JSON.parse(jsonText) : {} };
+            let jsonTextWithoutComments = removeComments(jsonText);
+            return { config: /\S/.test(jsonTextWithoutComments) ? JSON.parse(jsonTextWithoutComments) : {} };
         }
         catch (e) {
             return { error: createCompilerDiagnostic(Diagnostics.Failed_to_parse_file_0_Colon_1, fileName, e.message) };
         }
     }
 
+
+    /**
+     * Remove the comments from a json like text.
+     * Comments can be single line comments (starting with # or //) or multiline comments using / * * /
+     *
+     * This method replace comment content by whitespace rather than completely remove them to keep positions in json parsing error reporting accurate.
+     */
+    function removeComments(jsonText: string): string {
+        let result = "";
+        let processingString = false;
+        let processingSingleLineComment = false;
+        let processingMultiLineComment = false;
+        for (let i = 0; i < jsonText.length; i++) {
+            let currentCharCode = jsonText.charCodeAt(i);
+            let currentChar = jsonText.charAt(i);
+            let nextCharCode = (i + 1 < jsonText.length) ? jsonText.charCodeAt(i + 1) : undefined;
+            if (processingString) {
+                if (currentCharCode === CharacterCodes.backslash
+                    && nextCharCode !== undefined) {
+                    // Found an escaped character 
+                    // consume the \ and the escaped char
+                    result += currentChar;
+                    result += jsonText.charAt(i + 1);
+                    i += 1;
+                }
+                else if (currentCharCode === CharacterCodes.doubleQuote) {
+                    // End of string
+                    result += currentChar;
+                    processingString = false;
+                }
+                else {
+                   // String content
+                   result += currentChar;
+                }
+            }
+            else if (processingSingleLineComment) {
+                if (isLineBreak(currentCharCode)) {
+                    // End of single line comment
+                    processingSingleLineComment = false;
+                    // Keep the line breaks to keep line numbers aligned
+                    result += currentChar;
+                }
+                else {
+                   // replace comment content by whitespaces
+                   result += " ";
+                }
+            }
+            else if (processingMultiLineComment) {
+                if (currentCharCode === CharacterCodes.asterisk && nextCharCode === CharacterCodes.slash) {
+                    // End of comment */
+                    result += "  ";

```

Comment:
> This method replace comment content by whitespace rather than completely remove them to keep positions in json parsing error reporting accurate.

I was wondering the same thing, but then read the comment. Reasoning is good enough.